src/entity/posts/utils.ts

```diff
@@ -300,24 +372,29 @@ export const createSharePost = async (
 
     const id = await generateShortId();
 
-    const post = await con.getRepository(SharePost).save({
-      id,
-      shortId: id,
-      createdAt: new Date(),
-      sourceId,
-      authorId: userId,
-      sharedPostId: postId,
-      title: strippedCommentary,
-      titleHtml,
-      sentAnalyticsReport: true,
-      private: privacy,
-      origin: PostOrigin.UserGenerated,
-      visible,
-      visibleAt: visible ? new Date() : null,
-      flags: {
+    const post = await savePost<SharePost>({
+      con,
+      type: SharePost,
+      shouldUpdateFlags: true,
+      post: {
```

Comment:
Nothing was changed with the object we passed as `post`, just indentation due to being inside an object.